src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -161,6 +161,13 @@ public int SendTimeout
             }
         }
 
+        internal bool IsDisconnected { get; private set; } = false;
+
+        internal void SetToDisconnected()
+        {
+            IsDisconnected = true;
+        }
```

Comment:
Accessibility is an implementation detail; as long as it's not public/protected (which it's not), we can change it to meet our needs.

Lifetime is an issue, but I'm not understanding yet why the lifetime is different.  The purpose of these changes is to make the Unix behavior as close to the Windows behavior as possible.  Why do we need this tracked on SafeCloseSocket at all?  Can't we just use the tracking on Socket itself?  If reuseSocket is false, then the handle won't be replaced, and a subsequent attempt to connect will fail because the old handle that already had a connection attempt will still be there.

What am I missing?